### PR TITLE
Fix label selector in app values watcher

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/21118

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by tfussell — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/giantswarm/apiextensions-application v0.3.1
 	github.com/giantswarm/app/v6 v6.8.1
 	github.com/giantswarm/appcatalog v0.6.0
-	github.com/giantswarm/apptest v1.0.1
+	github.com/giantswarm/apptest v1.1.1
```
Comment (rossf7) [inline go.mod:11]: Fix for the test flakiness

File: integration/key/key.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func ChartOperatorUniqueName() string {
 }
 
 func ChartOperatorVersion() string {
-	return "2.13.1"
+	return "2.20.0"
```
Comment (rossf7) [inline integration/key/key.go:29]: Just bumping the chart-operator version used in tests as its old.

File: integration/setup/setup.go [hunk 3]
```diff
@@ -56,17 +58,32 @@ func installResources(ctx context.Context, config Config) error {
...
+		config.Logger.Debugf(ctx, "getting %#q tarball URL", project.Name())
+
+		o := func() error {
+			operatorTarballURL, err = appcatalog.GetLatestChart(ctx, key.ControlPlaneTestCatalogStorageURL(), project.Name(), env.CircleSHA())
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
```
Comment (rossf7) [inline integration/setup/setup.go:72]: A retry is needed here too. As we can't install app-operator as an app in the tests.

File: service/watcher/appvalue/appvalue.go [hunk 4]
```diff
@@ -42,14 +42,22 @@ func NewAppValueWatcher(config AppValueWatcherConfig) (*AppValueWatcher, error)
...
 	}
 
+	var selector labels.Selector
+	{
+		if config.WorkloadClusterID != "" {
+			selector = label.ClusterSelector(config.WorkloadClusterID)
+		} else {
+			selector = label.AppVersionSelector(config.UniqueApp)
+		}
+	}
```
Comment (rossf7) [inline service/watcher/appvalue/appvalue.go:52]: This is the actual fix.

**Pull request merged**